src/Dialogs/ColorPreferencesDialog.vala

```diff
@@ -0,0 +1,403 @@
+* modify it under the terms of the GNU Lesser General Public
+* License version 3, as published by the Free Software Foundation.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*/
+
+namespace Terminal.Dialogs {
+    public class ColorPreferences : Gtk.Dialog {
+
+        private Gtk.ListStore console_theme_store;
+        private Gtk.ComboBox console_theme_combobox;
+        private Gtk.Image contrast_image;
+
+        private Gtk.ColorButton color01_button; // Black
+        private Gtk.ColorButton color02_button; // Red
+        private Gtk.ColorButton color03_button; // Green
+        private Gtk.ColorButton color04_button; // Yellow
+        private Gtk.ColorButton color05_button; // Blue
+        private Gtk.ColorButton color06_button; // Magenta
+        private Gtk.ColorButton color07_button; // Cyan
+        private Gtk.ColorButton color08_button; // Light gray
+        private Gtk.ColorButton color09_button; // Dark gray
+        private Gtk.ColorButton color10_button; // Light Red
+        private Gtk.ColorButton color11_button; // Light Green
+        private Gtk.ColorButton color12_button; // Light Yellow
+        private Gtk.ColorButton color13_button; // Light Blue
+        private Gtk.ColorButton color14_button; // Light Magenta
+        private Gtk.ColorButton color15_button; // Light Cyan
+        private Gtk.ColorButton color16_button; // White
+        private Gtk.ColorButton background_button;
+        private Gtk.ColorButton foreground_button;
+        private Gtk.ColorButton cursor_button;
+
+        public signal void theme_changed ();
+
+        public ColorPreferences (Gtk.Window? parent) {
+            Object (
+                border_width: 6,
+                deletable: false,
+                resizable: false,
+                title: _("Color Preferences"),
+                transient_for: parent
+            );
+        }
+
+        construct {
+            get_content_area ().add (get_colors_box ());
+
+            var close_button = new Gtk.Button.with_label (_("Close"));
+            close_button.clicked.connect (destroy);
+
+            add_action_widget (close_button, 0);
+        }
+
+        private Gtk.Widget get_colors_box () {
+
+            var colors_grid = new Gtk.Grid () {
+                column_spacing = 12,
+                row_spacing = 6,
+                margin = 9,
+                margin_top = 0,
+                halign = Gtk.Align.CENTER
+            };
+
+            var window_theme_label = new SettingsLabel (_("Window Theme"));
```

Comment:
We have prior art for using the word "Style" instead of "Theme" when referring to the dark and light styles. I'm not sure if we have anything for referring to the app chrome specifically. Maybe @cassidyjames has suggestions here.

These labels should be sentence case as suffixed with ":"
```suggestion
            var window_theme_label = new SettingsLabel (_("Window style:"));
```